graylog2-server/src/main/java/org/graylog2/rest/resources/users/UsersResource.java

```diff
@@ -417,6 +420,17 @@ public void changeUser(@ApiParam(name = "userId", value = "The ID of the user to
                 user.setSessionTimeoutMs(sessionTimeoutMs);
             }
         }
+
+        if (user.isServiceAccount() && !cr.isServiceAccount()) {
+            // restore permissions that were removed from the service account user
+            final Set<String> selfEditPermissions = new Permissions(new HashSet<>()).userSelfEditPermissions(user.getName());
+            final List<String> userPermissions = user.getPermissions();
+            if (user.getPermissions().addAll(selfEditPermissions)) {
+                user.setPermissions(userPermissions);
+            }
+        }
```

Comment:
The self-edit permissions are added dynamically to a user at runtime in `UserImpl#getPermissions()` and are not stored in the database. `UserImpl#setPermissions()` also removes any self-edit permissions from the user, so the code is actually a no-op and should be removed.